### PR TITLE
Improve post creation fields

### LINKED ISSUE #130: Improve Posting Creation Request and Category fields
# Plan/Proposed Change
The Creation Screen currently has Switches for the Request and Category fields.
Either change these to dropdowns with the appropriate fields or add labels below switches to inform user what value they are selecting.

# Why is this important
Currently it is not completely clear what the values of the switches are. Need to make sure the user knows exactly what they are creating and the values associated.

# Additional Context/Questions
Values for Request are either: Offer or Request
Values for Category: Good or Service

To improve/facilitate searching by category, could provide a picker or TextInput for selecting a specific category for the posting to belong to. The Posting model on the backend already has a `category` field that takes a string value.

Might want to checkout the [react-native-picker](https://github.com/react-native-community/react-native-picker)

### PR DESCRIPTION
**Pull request by alexandra-hanson**
I changed the toggle switches on the "Create a posting" screen to picker select components, so that the user can select whether their post is a request or offer, or is a good or service. I also played with the layout of the fields on the page, so that they were a bit more spread out. I think there is still work to be done in formatting (particularly with the way the picker select components are laid out on the screen), but it should work functionally. 

- Tested only on Android so please try on iOS if you can

### CODE REVIEW

**Comment by seansisson** [comment]
Tested on Android (LG v20) and it looks great! I tested creating posts with various options and there are no anomalies. 

I noticed an interesting behavior that upon creating a new post, the options from the previous creation are saved, so if I created an offer and went to create a new post, that post would also default to offer unless otherwise specified. 

I think it's kinda cool that it does that, just something I noticed.

File: react-native/screens/PostingCreationScreen.js [hunk 1]
```diff
@@ -203,40 +217,63 @@ const PostingCreationScreen = props => {
...
+              </View>
+            <View>
               <View style={styles.countInputView}>
                 <TextInput
                   style={styles.countInputText}
+                  placeholderTextColor={Colors.placeholder_text}
                   keyboardType='numeric'
                   returnKeyType='done'
-                  value={itemCount.toString()}
+                  value = '1'
```
Comment (SimonBartonPSU) [inline react-native/screens/PostingCreationScreen.js]: This makes the quantity always reset to '1', at least on iOS .  I think it should be the `value={itemCount.toString()}`

**Comment by SimonBartonPSU** [comment]
Code looks excellent. Looks excellent on Droid. iOS is tricky with Pickers, may require some additional style props for iOS

Examples:

Posting creation begin
IMG-1706

Picker selected
IMG-1707

File: react-native/screens/PostingCreationScreen.js (hunk 1 above)
Comment (EHaake) [reply react-native/screens/PostingCreationScreen.js]: Nice catch.  That change fixes it.

**Comment by EHaake** [comment]
Just saw the iOS styling issues. Am working on it now. Will post updated screenshots when I think I have it looking better.

**Comment by EHaake** [comment]
Did some custom styling to improve look on iOS. Styling on android should be unchanged.

image0

**Comment by EHaake** [comment]
Also fixed issue with count value resetting with Simon's solution

**Review by SimonBartonPSU — APPROVED** [review]
Looks good!

**Pull request merged**